### ISSUE TITLE
Dates

### ISSUE THREAD

**Issue by SymbolixAU** [SymbolixAU/jsonify#15]
Move the date handling to C++, probably using `boost`, like what I'm attempting [here in colourvalues](https://github.com/SymbolixAU/colourvalues/blob/format/inst/include/colourvalues/format/format.hpp#L15)

**Comment by SymbolixAU** [comment]
ping @ChrisMuir I'd be interested in your thoughts. 

I've moved the Date handling code in `to_json()` from R, to c++. This means I can now handle dates/Posixct objects in nested lists too. However, at the moment it always converts the output to UTC. And I can't find an easy solution to keeping the time-zone attributes yet. 

Do you think the speed-gains, and functionality on lists, out-weights the changes in the output user sees?

**Comment by ChrisMuir** [comment]
Hey, sorry for the delay, wife and I just bought a house and we're in the middle of renovations/painting/packing/moving/etc, plus I'm starting a new job on Jan 7th, so things are kind of crazy right now. I haven't had a chance to look through the edits you've made, and I honestly don't have much experience with handling date-time objects. It looks like you merged edits to master though, going with the C++ approach. I'll take a look at everything when things settle down some.

**Comment by SymbolixAU** [comment]
yeah I merged it in to master so anyone who uses the dev version will see it and hopefully point out any issues. I think it's a better way of doing as it's faster and works on lists too. 

You may also see I've completely re-written the `to_json` API code. I'm going to be testing this over the next few days to see if it breaks anything in my other libraries, then I'll get onto the `from_json` branch.

Good luck with the move and settling in!
